chatgpt.sh

```diff
@@ -122,7 +122,10 @@ build_chat_context() {
 	else
 		chat_context="$chat_context\nQ: $escaped_request_prompt"
 	fi
-	request_prompt="${chat_context//$'\n'/\\n}"
+}
+
+escape(){
+  echo "$1" | jq -Rrs 'tojson[1:-1]'
```

Comment:
That is a very nice line of code. Could you share how we could do the opposite? Transform a json-escaped string (without the json structure) to raw output? So that echoing 'aaa\n\n' would result in "aa+two literal newlines"? Thanks.

Reply:
I asked ChatGPT about that but it didn't got it right from the first time so...

Here it is:

```
jq -Rr '"\"\(.)\"" | fromjson'
```

While the first direction takes the whole input as one string, converts it to its JSON representation and removes the surrounding double quotes, the inverse direction reads each line as a string adds double quotes around it and parse it from JSON.